source/common/router/upstream_request.cc

```diff
@@ -286,6 +295,11 @@ void UpstreamRequest::decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool e
 
   maybeHandleDeferredReadDisable();
   ASSERT(headers.get());
+
+  if (!end_stream) {
+    stream_info_.setStreamState(StreamInfo::StreamState::InProgress);
```

Comment:
docs in your prior PR indicate 
The value will be 'InProgress' in case an upstream connection was made,

This is being set on upstream headers being received which can be well after the stream is established. We'd also have to handle the case where we have an upstream stream, get an error response, and disassociate the stream for a pending retry  (ad may not get another immediae upstream connection).

I think we should scrap the state in stream info, and infer the successful upstream connection from the presence of an upstream host, making sure we set the state properly in access logging based on already existing stream info properties.

Reply:
@alyssawilk, I have deleted usage of StreamState in this PR. After this one is merged I'll create another one to adjust the code per what we discussed in the other PR. Can you please have another look?

Reply:
As I said above I think we need to work out the logging APIs before we land this. The release is cut next week and if we don't fix the logging before then we will have to either 1) roll back the other PR or 2) support the non-extensible APIs perpetually.  I would lean towards 1 so I do not want any dependant PRs landing.
/wait

Reply:
@alyssawilk can we merge this one, now that the revert PR is merged?